src/core/error_helpers.js

```diff
@@ -105,6 +110,159 @@ p5._friendlyFileLoadError = function (errorType, filePath) {
   report(message, errorInfo.method, FILE_LOAD);
 };
 
+/**
+* Validates parameters
+* param  {String}               func    the name of the function
+* param  {Array}                args    user input arguments
+*
+* example:
+*  var a;
+*  ellipse(10,10,a,5);
+* console ouput:
+*  "It looks like ellipse received an empty variable in spot #2."
+*
+* example:
+*  ellipse(10,"foo",5,5);
+* console output:
+*  "ellipse was expecting a number for parameter #1,
+*           received "foo" instead."
+*/
+function validateParameters(func, args) {
+  var arrDoc = lookupParamDoc(func);
+  if (arrDoc.length > 1){   // multiple format?
+    var errorListArray = [];
+    for (var i = 0; i < arrDoc.length; i++) {
+      errorListArray.push(testParamFormat(args, arrDoc[i]));
+    }
+    // compare errors from all formats
+    var minErrInd = -1;
+    var minErrCount = 0;
+    for (var j = 0; j < errorListArray.length; j++) {
+      var numErr = errorListArray[j].length;
+      if (numErr >= arrDoc.length){   // exclude clean cases
```

Comment:
a larger picture question here -- it seems to me that rather than checking the args against every possible param combo and then checking that at least one of them didn't fail, we could just invert this and just look for one that matches, and stop checking as soon as we find it. this would cut down checking time for functions that have many different possible arg combos. this doesn't need to be addressed before we merge, but it may be something to revisit later?